modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecDialog.kt

```diff
@@ -27,59 +30,130 @@ object MirrordExecDialog {
      *
      * @return a target selected from the given list, targetlessTargetName constant if user selected targetless, null if the user cancelled
      */
-    fun selectTargetDialog(podTargets: List<String>): String? {
-        val targets = podTargets.sorted().toMutableList()
-        MirrordSettingsState.instance.mirrordState.lastChosenTarget?.let {
-            val idx = targets.indexOf(it)
-            if (idx != -1) {
-                targets.removeAt(idx)
-                targets.add(0, it)
+    fun selectTargetDialog(availableTargets: List<String>): String? {
+        val targetsState = object {
+            private var _targets = availableTargets
+                    .sorted()
+                    .toMutableList()
+                    .apply {
+                        // Initially, the last used target is at the top
```

Comment:
imho, this should be a class called `TargetState` and you can just do `val targetsState = TargetsState(availableTargets)`

Reply:
Done, thanks!